tests/integration/local/invoke/test_integrations_cli.py

```diff
@@ -970,3 +1001,43 @@ def test_unresolved_layer_due_to_bad_instrinsic(self):
         expected_error_output = "Error: arn:aws:lambda:us-west-2:111111111101:layer:layerDoesNotExist:${LayerVersion} is an Invalid Layer Arn."
 
         self.assertIn(expected_error_output, error_output)
+
+
+class TestInvokeWithFunctionFullPathToAvoidAmbiguity(InvokeIntegBase):
+    template = Path("template-deep-root.yaml")
+    nested_stack_enabled = True
+
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_with_function_name_will_call_functions_in_top_level_stacks(self):
+        command_list = self.get_command_list("FunctionA", template_path=self.template_path, event_path=self.event_path)
```

Comment:
What will happen if I want to invoke `FunctionB` and it is only defined in one of the child stacks, would that work?
If so can we add the test here?

Reply:
Added two tests, one is `FunctionB` you mentioned. Another one is calling by function name instead of logical ID for function in a child stack.

Reply:
As discussed offline, added one more test to 

> Also call by FunctionSomeLogicalID in test_invoke_with_function_name_will_call_functions_in_top_level_stacks